src/ui/connectiondialog.cc

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2017 CodiLime
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include "ui/connectiondialog.h"
+#include "ui_connectiondialog.h"
+
+#include <climits>
+#include <cstdint>
+#include <ctime>
+#include <limits>
+#include <random>
+
+#include <QtGlobal>
+
+namespace veles {
+namespace ui {
+
+ConnectionDialog::ConnectionDialog(QWidget *parent)
+    : QDialog(parent),
+      ui_(new Ui::ConnectionDialog) {
+  ui_->setupUi(this);
+
+  connect(ui_->ok_button, &QPushButton::clicked, this, &QDialog::accept);
+  connect(ui_->cancel_button, &QPushButton::clicked, this, &QDialog::reject);
+  connect(ui_->server_localhost_button, &QPushButton::clicked,
+      this, &ConnectionDialog::serverLocalhost);
+  connect(ui_->client_localhost_button, &QPushButton::clicked,
+        this, &ConnectionDialog::clientLocalhost);
+  connect(ui_->random_key_button, &QPushButton::clicked,
+      this, &ConnectionDialog::randomKey);
+  connect(ui_->new_server_radio_button, &QPushButton::toggled,
+      this, &ConnectionDialog::newServerToggled);
+
+  db_file_dialog_ = new QFileDialog(this);
+  db_file_dialog_->setAcceptMode(QFileDialog::AcceptOpen);
+  db_file_dialog_->setFileMode(QFileDialog::AnyFile);
+
+  QStringList file_name_filters;
+  file_name_filters << "All files (*.*)";
+  db_file_dialog_->setNameFilters(file_name_filters);
+
+  connect(ui_->select_database_button, &QPushButton::clicked,
+        db_file_dialog_, &QFileDialog::show);
+  connect(db_file_dialog_, &QFileDialog::fileSelected,
+      this, &ConnectionDialog::databaseFileSelected);
+
+  server_file_dialog_ = new QFileDialog(this);
+  server_file_dialog_->setAcceptMode(QFileDialog::AcceptOpen);
+  server_file_dialog_->setFileMode(QFileDialog::ExistingFile);
```

Comment:
Currently, srv is an extensionless executable file.  I suppose we could change it...